src/main/scala/wdl4s/WdlNamespace.scala

```diff
@@ -143,76 +147,82 @@ case class WdlNamespaceWithWorkflow(importedAs: Option[String],
   */
 object WdlNamespace {
   def load(wdlFile: Path): WdlNamespace = {
-    load(readFile(wdlFile), wdlFile.toString, localImportResolver, None)
+    load(readFile(wdlFile), wdlFile.toString, fileResolver)
   }
 
   def load(wdlFile: Path, importResolver: ImportResolver): WdlNamespace = {
-    load(readFile(wdlFile), wdlFile.toString, importResolver, None)
+    load(readFile(wdlFile), wdlFile.toString, importResolver)
   }
 
   def load(wdlSource: WdlSource): WdlNamespace = {
-    load(wdlSource, "string", localImportResolver, None)
+    load(wdlSource, "string",  Seq(fileResolver), None)
+  }
+
+  def load(wdlSource: WdlSource, importResolver:ImportResolver): WdlNamespace = {
+    load(wdlSource, "string",  Seq(importResolver), None)
   }
 
-  def load(wdlSource: WdlSource, importResolver: ImportResolver): WdlNamespace = {
+  def load(wdlSource: WdlSource, importResolver: Seq[ImportResolver]): WdlNamespace = {
     load(wdlSource, "string", importResolver, None)
   }
 
   def load(wdlSource: WdlSource, resource: String): WdlNamespace = {
-    load(wdlSource, resource, localImportResolver, None)
+    load(wdlSource, resource, Seq(fileResolver), None)
   }
 
   def load(wdlSource: WdlSource, resource: String, importResolver: ImportResolver): WdlNamespace = {
-    load(wdlSource, resource, importResolver, None)
+    load(wdlSource, resource, Seq(importResolver), None)
   }
 
-  private def load(wdlSource: WdlSource, resource: String, importResolver: ImportResolver, importedAs: Option[String]): WdlNamespace = {
+  private def load(wdlSource: WdlSource, resource: String, importResolver: Seq[ImportResolver], importedAs: Option[String]): WdlNamespace = {
     WdlNamespace(AstTools.getAst(wdlSource, resource), wdlSource, importResolver, importedAs, root = true)
   }
 
-  def apply(ast: Ast, source: WdlSource, importResolver: ImportResolver, namespaceName: Option[String], root: Boolean = false): WdlNamespace = {
+
+  def apply(ast: Ast, source: WdlSource, importResolvers: Seq[ImportResolver], namespaceName: Option[String], root: Boolean = false): WdlNamespace = {
+
     val imports = for {
       importAst <- Option(ast).map(_.getAttribute("imports")).toSeq
       importStatement <- importAst.astListAsVector.map(Import(_))
     } yield importStatement
 
-    /**
-      * For import statements with an 'as' clause, this translates to a sub-namespace
-      */
-    val namespaces: Seq[WdlNamespace] = for {
-      imp <- imports if imp.namespaceName.isDefined
-      source = importResolver(imp.uri)
-      if source.length > 0
-    } yield WdlNamespace.load(source, imp.uri, importResolver, imp.namespaceName)
+
+    def tryResolve(str: String, remainingResolvers: Seq[ImportResolver], errors: List[Throwable]): Option[WdlSource] = {
```

Comment:
ToL: Add comment: `Make sure remainingResolvers has at least one entry the first time this is called`